feat(mongodb-atlas): add currency conversion support

malpou opened the pull request:
## What does this PR change?
* Adds a generic currency conversion package that can be reused across all OpenCost plugins
* Integrates currency conversion into the MongoDB Atlas plugin, allowing costs to be converted from USD to any ISO 4217 currency
* Implements caching with 24-hour TTL to minimize API calls (stays within free tier limit of 1,500 requests/month)
* Adds configuration options for target currency and exchange rate API key

## Does this PR relate to any other PRs?
* No

## How will this PR impact users?
* Users can now view MongoDB Atlas costs in their preferred currency by configuring `target_currency` and `exchange_api_key` in their plugin configuration
* No breaking changes - defaults to USD if not configured
* Other plugin developers can reuse the currency package for their own plugins

## Does this PR address any GitHub or Zendesk issues?
* No

## How was this PR tested?
* Unit tests for the currency package with mock implementations
* Integration tests for MongoDB Atlas plugin currency conversion

## Does this PR require changes to documentation?
* Yes - users need to know about the new configuration options:
  * `target_currency`: ISO 4217 currency code (e.g., "EUR", "GBP")
  * `exchange_api_key`: API key from exchangerate-api.com

malpou committed:
Add currency conversion support for MongoDB Atlas plugin

- Create generic currency package for reuse across plugins
- Implement exchange rate fetching from exchangerate-api.com
- Add thread-safe caching with 24-hour TTL
- Support all ISO 4217 currency codes
- Update MongoDB Atlas plugin to support currency conversion
- Add configuration for target currency and API key
- Include comprehensive tests for currency conversion

Signed-off-by: Malthe Poulsen <malthe@grundtvigsvej.dk>

malpou committed:
don't initialize to default TTL value

Signed-off-by: Malthe Poulsen <malthe@grundtvigsvej.dk>

malpou committed:
Update module references for `common/currency` package

Signed-off-by: Malthe Poulsen <malthe@grundtvigsvej.dk>

malpou committed:
Refactor currency handling by introducing `defaultCurrency` constant and replacing hardcoded "USD" references.

Signed-off-by: Malthe Poulsen <malthe@grundtvigsvej.dk>

malpou committed:
Standardize naming conventions across currency package components. Use unexported types/fields and camelCase for uniformity. Update associated tests and methods accordingly.

Signed-off-by: Malthe Poulsen <malthe@grundtvigsvej.dk>

ameijer commented:
@malpou would you be willing to move into opencost, then invoke the APIs from there? we would love to bring this functionality to all parts of the product

malpou commented:
@ameijer Yes of course. 

If you can give a little guidance to where and how it should land in the opencost project then I'd move it in there 😄

ameijer commented:
Could I suggest you move your 'currency' package to https://github.com/opencost/opencost/tree/develop/pkg? I think that is a logical home for it

malpou committed:
refactor: Update MongoDB Atlas plugin to use currency package from opencost

- Remove local currency package from opencost-plugins/common/currency
- Update imports to use github.com/opencost/opencost/pkg/currency
- Fix duplicate imports and incorrect replace directives in go.mod
- Remove unused standard library plugin import

Signed-off-by: malpou <malthe@grundtvigsvej.dk>

ameijer commented on pkg/common/currency/cache.go:
```diff
@@ -0,0 +1,99 @@
+package currency
```
@malpou can we update this to remove currency conversion since we merged your PR upstream?

malpou replied:
Sorry for the long time of nothing happening been busy, @ameijer it should be good now 😄
